### PR TITLE
chore(Tooltip): converted to int test

### LINKED ISSUE #1296: Tooltip: Convert to integration test
Currently this is still using the old method of 'newing' up the component class. We should convert this to use the TestBed and fully test it.

Might have to add a toggle (material has one, we don't) to show/hide the tooltip, even if it's just for testing purposes.

### PR DESCRIPTION
**Pull request by shani-terminus**
ISSUES CLOSED: #1296

### CODE REVIEW

File: terminus-ui/tooltip/src/tooltip.component.spec.ts [hunk 1]
```diff
@@ -1,46 +1,139 @@
...
-      component.tooltipPosition = 'above';
-      expect(component.tooltipPosition).toEqual('above');
+    describe(`tooltipPosition`, () => {
+      test(`should set and retrieve above`, () => {
+        component.tooltipPosition = 'above';
+        fixture.detectChanges();
 
-    });
+        expect(tooltipComponent.tooltipPosition).toEqual('above');
+        expect(tooltipEl.attributes.getNamedItem('ng-reflect-position').value).toEqual('above');
```
Comment (benjamincharity) [inline terminus-ui/tooltip/src/tooltip.component.spec.ts]: I don't think we should be testing `ng-*` data attributes directly. We have no control over those values. We should probably be testing for positions or classes. (same comment below)

File: terminus-ui/tooltip/src/tooltip.component.html [hunk 2]
```diff
@@ -1,8 +1,9 @@
-<span
+<span #tooltip="matTooltip"
   class="c-tooltip qa-tooltip"
   [matTooltip]="tooltipValue"
   [matTooltipPosition]="tooltipPosition"
   [ngClass]="{'c-tooltip--underline': hasUnderline}"
+  (click)="tooltip.show()"
```
Comment (benjamincharity) [inline terminus-ui/tooltip/src/tooltip.component.html]: We shouldn't be changing how the tooltip works for consumers. I'd rather add a `show` method to our tooltip component that calls Material's `show`.

File: terminus-ui/tooltip/testing/src/test-helpers.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { ComponentFixture } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+
+/**
+ * Helper function to display the tooltip
+ *
+ * @param fixture - the fixture with a tooltip
+ */
+// tslint:disable-next-line:no-any
+export function showTooltip(fixture: ComponentFixture<any>) {
```
Comment (benjamincharity) [inline terminus-ui/tooltip/testing/src/test-helpers.ts]: Need `void` return type here

**Review by benjamincharity — CHANGES_REQUESTED** [review]

File: terminus-ui/tooltip/testing/src/test-helpers.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { ComponentFixture } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { MatTooltip } from '@angular/material';
```
Comment (benjamincharity) [inline terminus-ui/tooltip/testing/src/test-helpers.ts]: I don't think this one is being used anymore

File: terminus-ui/tooltip/testing/src/test-helpers.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+
+// @ViewChild(MatTooltip) tooltip:MatTooltip;
```
Comment (benjamincharity) [inline terminus-ui/tooltip/testing/src/test-helpers.ts]: Legacy comment

File: terminus-ui/tooltip/testing/src/test-components.ts [hunk 6]
```diff
@@ -0,0 +1,69 @@
+import {
+  Component,
+  NgModule,
+  ViewChild,
+} from '@angular/core';
+
+import {
+  TsTooltipComponent,
+  TsTooltipModule,
+} from '../../src/tooltip.module';
```
Comment (benjamincharity) [inline terminus-ui/tooltip/testing/src/test-components.ts]: This import should be like an external consumer: `@terminus/ui/tooltip`

File: terminus-ui/tooltip/src/tooltip.component.ts [hunk 7]
```diff
@@ -79,4 +83,17 @@ export class TsTooltipComponent {
...
+  /**
+   * Access Material Tooltip Directive
+   */
+  @ViewChild('tooltip', {static: true})
+  public matTooltip;
+
+  /**
+   * Call Material Tooltip's show() method
+   */
+  public showTooltip() {
```
Comment (benjamincharity) [inline terminus-ui/tooltip/src/tooltip.component.ts]: Missing void return type

File: terminus-ui/tooltip/src/tooltip.component.ts [hunk 8]
```diff
@@ -3,8 +3,12 @@ import {
   Component,
   Input,
   isDevMode,
+  ViewChild,
   ViewEncapsulation,
 } from '@angular/core';
+import { MatTooltip } from '@angular/material';
```
Comment (benjamincharity) [inline terminus-ui/tooltip/src/tooltip.component.ts]: I don't think this one is being used anymore

**Review by benjamincharity — CHANGES_REQUESTED** [review]

File: terminus-ui/button/src/button.component.spec.ts [hunk 9]
```diff
@@ -7,6 +7,7 @@ import {
   tick,
 } from '@angular/core/testing';
 import { By } from '@angular/platform-browser';
+import { getDomAttribute } from '@terminus/ngx-tools';
```
Comment (benjamincharity) [inline terminus-ui/button/src/button.component.spec.ts]: Oh, also, this is a rebase issue I think.. you can go ahead and remove this line.

File: terminus-ui/tooltip/testing/src/test-components.ts (hunk 6 above)
Comment (shani-terminus) [reply terminus-ui/tooltip/testing/src/test-components.ts]: Yes, this was supposed to be temporary

**Review by benjamincharity — APPROVED** [review]

**Pull request merged**